src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -128,7 +128,7 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_context_t,
         /* Global ctx mode, we will have ctx_map so can use array for eps */
         self->eps = ucc_calloc(params->context->addr_storage.size,
                                sizeof(ucp_ep_h), "ucp_eps");
-        if (self->eps) {
+        if (!self->eps) {
```

Comment:
wow ... how does the code work today? global ctx mode is default in almost all the tests. will check.